Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -0,0 +1,546 @@
+{
+    std::vector<std::string> tokens;
+    std::stringstream sstream(input);
+    std::copy(std::istream_iterator<std::string>(sstream), std::istream_iterator<std::string>(), std::back_inserter(tokens));
+
+    if (tokens.size() < 1) {
+        return true;
+    }
+
+    std::string command = tokens[0];
+    if (command == "Print") {
+        std::queue<std::string> keyPath;
+        if (tokens.size() > 1) {
+            ParseCommandKeyPathString(tokens[1], &keyPath);
+        }
+        Print(root.object.get(), keyPath, xml);
+    } else if (command == "Save") {
+        std::queue<std::string> keyPath;
+        Print(root.object.get(), keyPath, xml, filesystem, path);
+    } else if (command == "Exit") {
+        return false;
+    } else if (command == "Set") {
+        if (tokens.size() < 2) {
+            fprintf(stderr, "Set command requires KeyPath\n");
+            return true;
+        } else {
+            std::queue<std::string> keyPath;
+            ParseCommandKeyPathString(tokens[1], &keyPath);
+            Set(root.object.get(), keyPath, plist::ObjectType::String, parseCommandValueString(tokens.begin() + 2, tokens.end()));
+        }
+    } else if (command == "Add") {
+        if (tokens.size() < 3) {
+            fprintf(stderr, "Add command requires KeyPath and Type\n");
+            return true;
+        } else {
+            std::queue<std::string> keyPath;
+            ParseCommandKeyPathString(tokens[1], &keyPath);
+            plist::ObjectType type = ParseType(tokens[2]);
+            Set(root.object.get(), keyPath, type, parseCommandValueString(tokens.begin() + 3, tokens.end()), false);
+        }
+    } else if (command == "Clear") {
+        plist::ObjectType clearType;
+        if (tokens.size() < 2) {
+            clearType = plist::ObjectType::Dictionary;
+        } else {
+            clearType = ParseType(tokens[1]);
+        }
+        Clear(root, clearType);
+    } else if (command == "Delete") {
+        if (tokens.size() < 2) {
+            fprintf(stderr, "Add command requires KeyPath\n");
+            return true;
+        }
+        std::queue<std::string> keyPath;
+        ParseCommandKeyPathString(tokens[1], &keyPath);
+        Delete(root.object.get(), keyPath);
+    } else if (command == "Help") {
+        CommandHelp();
+    } else {
+        fprintf(stderr, "Unrecognized command\n");
+    }
+    return true;
+}
+
+int
+main(int argc, char **argv)
+{
+    std::vector<std::string> args = std::vector<std::string>(argv + 1, argv + argc);
+    auto filesystem = libutil::DefaultFilesystem();
+
+    Options options;
+    std::pair<bool, std::string> result = libutil::Options::Parse<Options>(&options, args);
+    if (!result.first) {
+        return Help(result.second);
+    }
+
+    if (options.help() || options.input().empty()) {
+        Help();
+        return 1;
```

Comment:
Well, this path is taken also when user types in "Help"
In either case, it is kind of self evident, and not sure if breaking this into two separate if's is worth.